doc/source/tutorial/linalg_batch.md

```diff
@@ -0,0 +1,148 @@
+batch
+```
+
+We could perform the operation for each element of the batch in a loop or list comprehension:
+
+```{code-cell} ipython3
+[linalg.det(A) for A in batch]
+```
+
+However, just as we might use NumPy broadcasting and vectorization rules to create the batch of matrices in the first place:
+
+```{code-cell} ipython3
+i = np.arange(1, 4).reshape(-1, 1, 1)
+batch = i * np.eye(3)
+batch
+```
+
+we might also wish to perform the determinant operation on all of the matrices in one function call.
+
+```{code-cell} ipython3
+linalg.det(batch)
+```
+
+In SciPy, we prefer the term "batch" instead of "stack" because the idea is generalized to N-dimensional batches. Suppose the input is a 2 x 4 batch of 3 x 3 matrices.
+
+```{code-cell} ipython3
+batch_shape = (2, 4)
+i = np.arange(np.prod(batch_shape)).reshape(*batch_shape, 1, 1)
+input = i * np.eye(3)
+```
+
+In this case, we say that the *batch shape* is `(2, 4)`, and the *core shape* of the input is `(3, 3)`. The net shape of the input is the sum (concatenation) of the batch shape and core shape.
+
+```{code-cell} ipython3
+input.shape
+```
+
+Since each 3 x 3 matrix is converted to a zero-dimensional scalar, we say that the core shape of the outuput is `()`. The shape of the output is the sum of the batch shape and core shape, so the result is a 2 x 4 array.
+
+```{code-cell} ipython3
+output = linalg.det(input)
+output
+```
+
+```{code-cell} ipython3
+output.shape
+```
+
+Not all linear algebra functions map to scalars. For instance, the `linalg.expm` function maps from a matrix to a matrix with the same shape.
+
+```{code-cell} ipython3
+A = np.eye(3)
+linalg.expm(A)
+```
+
+In this case, the core shape of the output is `(3, 3)`, so with a batch shape of `(2, 4)`, we expect an output of shape `(2, 4, 3, 3)`.
+
+```{code-cell} ipython3
+output = linalg.expm(input)
+output.shape
+```
+
+Generalization of these rules to functions with multiple inputs and outputs is straightforward. For instance, the `linalg.eig` function produces two outputs by default, a vector and a matrix.
+
+```{code-cell} ipython3
+evals, evecs = linalg.eig(A)
+evals.shape, evecs.shape
+```
+
+In this case, the core shape of the output vector is `(3,)` and the core shape of the output matrix is `(3, 3)`. The shape of each output is the batch shape plus the core shape as before.
+
+```{code-cell} ipython3
+evals, evecs = linalg.eig(input)
+evals.shape, evecs.shape
+```
+
+When there is more than one input, there is no complication if the input shapes are identical.
+
+```{code-cell} ipython3
+evals, evecs = linalg.eig(input, b=input)
+evals.shape, evecs.shape
+```
+
+The rules when the shapes are not identical follow logically. Each input can have its own batch shape as long as the shapes are broadcastable according to NumPy's broadcasting rules. The net batch shape is the broadcasted shape of the individual batch shapes, and the shape of each output is the net batch shape plus its core shape.
```

Comment:
Can I link to NumPy's broadcasting rules with the normal markdown syntax?

Reply:
Yes, including intersphinx! So you can do 

```
The rules when the shapes are not identical follow logically. Each input can have its own
batch shape as long as the shapes are broadcastable according to
[NumPy's broadcasting rules](#array-broadcasting-in-numpy).
The net batch shape is the broadcasted shape of the individual batch shapes, and the shape
of each output is the net batch shape plus its core shape.
```

where `array-broadcasting-in-numpy` is the label in the broadcasting guide for NumPy:

https://github.com/numpy/numpy/blob/f3b3dc536e624b20822cd37cd01c15e3767af84a/doc/source/user/basics.broadcasting.rst?plain=1#L2

Reply:
Ah, great. Will do.